spec-main/extensions-spec.ts

```diff
@@ -381,6 +390,62 @@ describe('chrome extensions', () => {
       expect(response.message).to.equal('Hello World!');
       expect(response.tabId).to.equal(w.webContents.id);
     });
+
+    it('get returns correct details', async () => {
+      const customSession = session.fromPartition(`persist:${uuid.v4()}`);
+      const extension = await customSession.loadExtension(path.join(fixtures, 'extensions', 'chrome-api'));
+      const w = new BrowserWindow({ show: false, webPreferences: { session: customSession, nodeIntegration: true } });
+      await w.loadURL(`${extension.url}/index.html`);
+
+      customSession.setExtensionAPIHandlers({
+        getTab: () => {
+          return {
+            groupId: 2
+          };
+        }
+      });
+
+      const tab = await w.webContents.executeJavaScript(`new Promise((resolve) => { chrome.tabs.get(${w.webContents.id}, resolve) })`);
+
+      expect(tab.groupId).to.equal(2);
+    });
+
+    it('get does not return a tab even if it exists', async () => {
+      const customSession = session.fromPartition(`persist:${uuid.v4()}`);
+      const extension = await customSession.loadExtension(path.join(fixtures, 'extensions', 'chrome-api'));
+      const w = new BrowserWindow({ show: false, webPreferences: { session: customSession, nodeIntegration: true } });
+      await w.loadURL(`${extension.url}/index.html`);
+
+      customSession.setExtensionAPIHandlers({
+        getTab: () => {
+          return null;
+        }
+      });
+
+      const tab = await w.webContents.executeJavaScript(`new Promise((resolve) => { chrome.tabs.get(${w.webContents.id}, resolve) })`);
+
+      expect(tab).to.be.undefined();
+    });
+
+    it('copes with the webcontents being destroyed', async () => {
+      const customSession = session.fromPartition(`persist:${uuid.v4()}`);
+      const extension = await customSession.loadExtension(path.join(fixtures, 'extensions', 'chrome-api'));
+      const w = new BrowserWindow({ show: false, webPreferences: { session: customSession, nodeIntegration: true } });
+      await w.loadURL(`${extension.url}/index.html`);
+
+      const p = new Promise(resolve => {
+        customSession.setExtensionAPIHandlers({
+          getTab: (wc) => {
+            setTimeout(resolve);
+            wc.destroy();
+            return null;
+          }
+        });
+      });
+
+      w.webContents.executeJavaScript(`new Promise((resolve) => { chrome.tabs.get(${w.webContents.id}, resolve) })`);
+      await p;
+    });
```

Comment:
I feel like there should be some more tests here. e.g. I don't think we test the behavior of `getActiveTab`?

Reply:
Actually the executeScript test uses `getActiveTab` here: https://github.com/electron/electron/pull/27917/files#diff-091ff3a664b65a57fb6555e5651d0079ab18aa70a8ca26a12eda85bb759194f1R348

Reply:
@sentialx I'd prefer to see a more targeted test that doesn't just incidentally happen to call the method under test. e.g. it should test that you can return `null` from the handler, should assert that the handler was called, that the correct webContents was returned, etc.

Reply:
@nornagon Sorry, I forgot about this. I can't see any other way of testing it, since nothing actually calls the getActiveTab callback besides `executeScript`, `getZoom` etc. Should I just copy-paste the test with executeScript and change the name of the test?